crates/secrets/src/aws_secrets_manager.rs

```diff
@@ -32,12 +50,99 @@ impl AwsSecretsManager {
     pub fn new() -> Self {
         Self {}
     }
+
+    /// Initializes AWS configuration and verifies AWS credentials.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error:
+    /// - If the AWS configuration cannot be loaded.
+    /// - If the call to STS `get_caller_identity` fails, which might be due to invalid or expired AWS credentials.
+    pub async fn init(self) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
We need to use Snafu errors:
https://github.com/spiceai/spiceai/blob/trunk/docs/dev/style_guide.md#all-errors-use-snafu-functionality

Reply:
Updated, thank you @phillipleblanc!